### PR TITLE
Add plugin documentation page

### PR DESCRIPTION
**Pull request by BryceGattis**
Need some help ironing this page out, but wanted to get a doc page started for this.

### CODE REVIEW

File: docs/source/plugins.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+=======
+
+Plugins are extensions of rez that extend rez functionality.
+
+Plugins are currently bundled in the main rez repo, but will be split out
+to their own repos in the future.
+
+The plugins that exist are currently located at :gh-rez:`src/rezplugins`.
+
+Existing plugins
```
Comment (JeanChristopheMorinPerso) [inline docs/source/plugins.rst:12]: These are plugin types, not plugins.

File: docs/source/plugins.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- :gh-rez:`src/rezplugins/build_process/shell`
+
+Developing your own plugin
+==========================
+
+Rez plugins require a specific folder structure as follows:
+
+.. code-block:: text
+
+    /plugin_name
```
Comment (JeanChristopheMorinPerso) [inline docs/source/plugins.rst:30]: This is a plugin type folder. Things inside that folder are the actual plugins.

File: docs/source/plugins.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
+=======
+Plugins
+=======
+
+Plugins are extensions of rez that extend rez functionality.
+
+Plugins are currently bundled in the main rez repo, but will be split out
+to their own repos in the future.
+
+The plugins that exist are currently located at :gh-rez:`src/rezplugins`.
```
Comment (JeanChristopheMorinPerso) [inline docs/source/plugins.rst:10]:
```suggestion
The built-in plugins are located at :gh-rez:`src/rezplugins`.
```

File: docs/source/plugins.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
+=======
+Plugins
+=======
+
+Plugins are extensions of rez that extend rez functionality.
```
Comment (JeanChristopheMorinPerso) [inline docs/source/plugins.rst:5]:
```suggestion
Rez is designed around the concept of plugins. Plugins can be used to extend rez's functionalities without modifying any of rez's source code.
```

File: docs/source/plugins.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+Plugin types
+------------
+
+There are two different plugin types in Rez.
+
+- Plugin
+- Extension
+
+Extensions differ from standard plugins in that they add a new command to rez's
+CLI.
```
Comment (JeanChristopheMorinPerso) [inline docs/source/plugins.rst:50]: I'm not too sure about this section. "extensions" are not special from what I know. We could explain what an extension is by saying that a plugin that adds a command is sometimes refereed to as an extension?

**Review by JeanChristopheMorinPerso — COMMENTED** [review]
Thanks! It's already a great start. I left some comments.

Also, it would be nice to cover these things:

* How are plugins configured?
* How can we install an are out-of-tree plugin? (out-of-tree here means a plugin that lives outside rez's source code)
* How can we know which plugins are currently loaded? (is this even possible without using the API?)

File: docs/source/plugins.rst (hunk 5 above)
Comment (aclark4life) [reply docs/source/plugins.rst:50]: Yeah I would drop "extension" entirely if possible. Plugins exist. You can do a lots of things with plugins, including adding `rez` commands. In fact, it's impossible to add a plugin without "extending" rez's functionality in some way so calling only plugins that add `rez` commands "extensions" is arguably inaccurate, or incorrect, by some folks' definition.

File: docs/source/plugins.rst [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+Required file contents
+----------------------
+``__init__.py``
+
+.. code-block:: python
```
Comment (JeanChristopheMorinPerso) [inline docs/source/plugins.rst:56]:
```suggestion

.. code-block:: python
   :caption: __init__.py
```
I think the `caption` option will add a title to the code block.

File: docs/source/plugins.rst [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+   shell               cmd         Windows Command Prompt (DOS) shell.                loaded
+   shell               gitbash     Git Bash (for Windows) shell                       loaded
+   shell               powershell  Windows PowerShell 5                               loaded
+   shell               pwsh        PowerShell Core 6+                                 loaded
+
+
+Configuring plugins
+===================
+
+# TODO: Write this part.
```
Comment (BryceGattis) [inline docs/source/plugins.rst:48]: Tried poking through some code but AFAIK you can't actually configure plugins in your own `rezconfig.py` outside of the repo structure, but maybe I'm missing something? Can you help me out here @JeanChristopheMorinPerso?

The whole idea that you have to configure plugin settings in two different places in the repo, the "plugin type" settings at the root of the repo and then specific plugin settings in the `rezplugins` folder is pretty odd to me.

Comment (JeanChristopheMorinPerso) [reply docs/source/plugins.rst:48]: You can configure plugins in your own rezconfig.py like so:

```python
plugins = {
    "filesystem": {}
}
```

Comment (JeanChristopheMorinPerso) [reply docs/source/plugins.rst:48]: See https://github.com/AcademySoftwareFoundation/rez/blob/main/src%2Frez%2Fconfig.py#L643-L648

File: docs/source/plugins.rst [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+Plugins are currently bundled in the main rez repo, but will be split out
+to their own repos in the future.
+
+The built-in plugins are located at :gh-rez:`src/rezplugins`.
+
+Loaded plugins
+==============
+
+Currently loaded plugins can be queried by running ``rez -i``
```
Comment (JeanChristopheMorinPerso) [inline docs/source/plugins.rst:15]:
```suggestion
Currently loaded plugins can be queried by running :option:`rez -i`
```

File: docs/source/plugins.rst [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+=======
+
+Rez is designed around the concept of plugins. Plugins can be used to extend rez's functionalities without modifying any of rez's source code.
+
+Plugins are currently bundled in the main rez repo, but will be split out
+to their own repos in the future.
+
+The built-in plugins are located at :gh-rez:`src/rezplugins`.
+
+Loaded plugins
```
Comment (JeanChristopheMorinPerso) [inline docs/source/plugins.rst:12]: View active plugins? List installed plugins?

File: docs/source/plugins.rst [hunk 10]
```diff
@@ -0,0 +1,137 @@
+=======
+Plugins
+=======
+
+Rez is designed around the concept of plugins. Plugins can be used to extend rez's functionalities without modifying any of rez's source code.
+
+Plugins are currently bundled in the main rez repo, but will be split out
+to their own repos in the future.
+
+The built-in plugins are located at :gh-rez:`src/rezplugins`.
```
Comment (JeanChristopheMorinPerso) [inline docs/source/plugins.rst:10]:
```suggestion
Rez comes with built-in plugins that are located at :gh-rez:`src/rezplugins`. New plugins are encouraged to be developed out-of-tree (outside rez).
```
It is true that we want to maybe split them out (at least some of them), but we didn't make any decision yet on this.

**Review by JeanChristopheMorinPerso — APPROVED** [review]
I'm quite happy with the result. Thanks again @BryceGattis!

**Pull request merged**